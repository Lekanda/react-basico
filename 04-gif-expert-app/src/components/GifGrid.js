import React from 'react'

export const GifGrid = ({category}) => {

    const getGifs =async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morti&limit=10&api_key=Fy5o4SLF3Q2ij74a1bG1N3NoL20R8Cv8';

        const resp = await fetch(url);
        const {data} = await resp.json();
        
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    getGifs();

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}
