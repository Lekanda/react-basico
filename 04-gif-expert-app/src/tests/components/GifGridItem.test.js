import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';



describe('Pruebas en GifGridItem', () => {
    const title = 'Titulo';
    const url = 'https://localhost/img.jpg';
    
    test('Debe mostrar el <GifGridItem/>', () => {
        const wrapper = shallow(<GifGridItem title={title} url={url}/>);
        
        expect(wrapper).toMatchSnapshot();
    });
})

