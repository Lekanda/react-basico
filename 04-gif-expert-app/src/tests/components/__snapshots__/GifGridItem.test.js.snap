// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrar el <GifGridItem/> 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__slow"
>
  <img
    alt="Titulo"
    src="https://localhost/img.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
